runners/flink/flink_runner.gradle

```diff
@@ -137,11 +138,9 @@ dependencies {
   compile project(":runners:java-job-service")
   compile project(":sdks:java:extensions:google-cloud-platform-core")
   compile library.java.vendored_grpc_1_26_0
-  compile library.java.jackson_annotations
   compile library.java.slf4j_api
   compile library.java.joda_time
   compile library.java.args4j
-  compile "org.apache.flink:flink-clients_2.11:$flink_version"
```

Comment:
The precommit is catching a legitimate error. I think flink-clients needs to be on the runtime classpath in order to provide Flink local mode. Try making flink-clients `runtimeOnly`.

```
org.apache.beam.sdk.Pipeline$PipelineExecutionException: java.lang.IllegalStateException: No ExecutorFactory found to execute the application.
	at org.apache.beam.runners.flink.TestFlinkRunner.run(TestFlinkRunner.java:75)
```

Reply:
@ibzib all checks are passing now. Do you think this is good to go?

Reply:
@ibzib I noticed this while testing the 2.28.0 SNAPSHOTs. Now from the user side I am forced to pass this dependency. Is this a regression? if this is intended (I doubt) we should probably announce it in CHANGES.md